R/reportingengine-env.R

```diff
@@ -250,3 +249,135 @@ setDefaultAutoAxisLimitMargin <- function(margin) {
+#' \item `"5th-95th Percentiles"` summarizes data using median, 5th and 95th percentiles
+#' \item `"10th-90th Percentiles"` summarizes data using median, 10th and 90th percentiles
+#' \item `"Geometric mean"` summarizes data using geometric mean and mean */ geometric standard deviation
+#' \item `"Arithmetic mean"` summarizes data using arithmetic mean and mean +/- standard deviation
+#' }
+#' 
+#' @export
+#' @family enum helpers
+#' @examples 
+#' \dontrun{
+#' getStatisticsFromType(StatisticsTypes)
+#' }
+StatisticsTypes <- enum(c(
+  "2.5th-97.5th Percentiles",
+  "5th-95th Percentiles",
+  "10th-90th Percentiles",
+  "Geometric mean",
+  "Arithmetic mean"
+))
+
+
+#' @title getStatisticsFromType
+#' @description Get statistics 
+#' @param statisticsType Statistics summarizing time profile simulated data
+#' as defined by helper enum `StatisticsType`
+#' @return A list including `y`, `ymin` and `ymax` summary statistics as well as their `caption`
+#' @export
+#' @examples 
+#' \dontrun{
+#' workflow$plotTime
+#' }
+getStatisticsFromType <- function(statisticsType){
+  validateIsIncluded(statisticsType, StatisticsTypes)
+  if(isIncluded(statisticsType, StatisticsTypes$`2.5th-97.5th Percentiles`)){
+    return(list(
+      y = tlf::tlfStatFunctions$`Percentile50%`,
+      ymin = tlf::tlfStatFunctions$`Percentile2.5%`,
+      ymax = tlf::tlfStatFunctions$`Percentile97.5%`,
+      yCaption = "median",
+      # The unicode characters below are superscript th
+      rangeCaption = "[2.5\u1d57\u02b0-97.5\u1d57\u02b0] percentiles"
+    ))
+  }
+  if(isIncluded(statisticsType, StatisticsTypes$`5th-95th Percentiles`)){
+    return(list(
+      y = tlf::tlfStatFunctions$`Percentile50%`,
+      ymin = tlf::tlfStatFunctions$`Percentile5%`,
+      ymax = tlf::tlfStatFunctions$`Percentile95%`,
+      yCaption = "median",
+      rangeCaption = "[5\u1d57\u02b0-95\u1d57\u02b0] percentiles"
+    ))
+  }
+  if(isIncluded(statisticsType, StatisticsTypes$`10th-90th Percentiles`)){
+    return(list(
+      y = tlf::tlfStatFunctions$`Percentile50%`,
+      ymin = tlf::tlfStatFunctions$`Percentile10%`,
+      ymax = tlf::tlfStatFunctions$`Percentile90%`,
+      yCaption = "median",
+      rangeCaption = "[10\u1d57\u02b0-90\u1d57\u02b0] percentiles"
+    ))
+  }
+  if(isIncluded(statisticsType, StatisticsTypes$`Arithmetic mean`)){
+    return(list(
+      y = tlf::tlfStatFunctions$mean,
+      ymin = tlf::tlfStatFunctions$`mean-sd`,
+      ymax = tlf::tlfStatFunctions$`mean+sd`,
+      yCaption = "arithmetic mean",
+      # The unicode character below is +/- symbol
+      rangeCaption = "mean \u00b1 SD range"
+    ))
+  }
+  #TODO : define geometric mean in tlf !
+  return(list(
+    y = "geomean",
+    ymin = "geomeanDividedBySD",
+    ymax = "geomeanMultipliedBySD",
+    yCaption = "geometric mean",
+    # The unicode character below is supposed to be */ symbol
+    rangeCaption = "mean \u22c7 geometric SD range"
```

Comment:
was the caption supposed to be "mean */ geometric SD range"?
because when entering this in R I get:
```
> "mean \u22c7 geometric SD range"
[1] "mean ⋇ geometric SD range"
```

Reply:
Let's use "*/".
After looking at the symbol in very big resolution (https://www.compart.com/en/unicode/U+22c7) I finally understood what it is, but in the smaller size it just looks like "*"